api/src/owner_rpc.rs

```diff
@@ -363,6 +367,14 @@ impl OwnerRpc for Owner {
 		Owner::validate_chain(self).map_err(|e| e.kind().clone())
 	}
 
+	fn reset_chain_head(&self, hash: String) -> Result<(), ErrorKind> {
```

Comment:
Do calls to this API endpoint require a password / extra authentication?

The attack vector would be for a malicious caller to send a hash to their chain head.

Reply:
Yes this endpoint  is exposed via the "owner" api alongside ban/unban peer endpoints etc.